lia-ui/src/main/java/com/lithium/community/android/ui/components/adapters/LiConversationAdapter.java

```diff
@@ -169,136 +185,318 @@ private void showPopupMenu(final View view, final int position) {
+        boolean isSubscribed = item.getUserContext().isSubscribed();
+        popup.getMenu().findItem(R.id.li_action_subscribe).setVisible(showSubscribe);
+        popup.getMenu().findItem(R.id.li_action_subscribe).setTitle(isSubscribed ? R.string.li_unsubscribe : R.string.li_subscribe);
+        popup.setOnMenuItemClickListener(new PopupMenuClickListener(item, getItems().size() == 1 || position == 0, position));
+        popup.show();
+    }
+
+    private void actionInProgress(boolean inProgress) {
+        if (mActionInProgressListener != null) {
+            mActionInProgressListener.actionInProgress(inProgress);
+        }
+    }
+
+    private class PopupMenuClickListener implements PopupMenu.OnMenuItemClickListener {
+
+        private final LiMessage item;
+        private final boolean isParentItem;
+        private final int position;
+
+        PopupMenuClickListener(LiMessage item, boolean isParentItem, int position) {
+            this.isParentItem = isParentItem;
+            this.position = position;
+            this.item = item;
+        }
+
+        @Override
+        public boolean onMenuItemClick(MenuItem menuItem) {
+            if (menuItem.getItemId() == R.id.li_action_report_abuse) {
+                openReportAbuse(item);
+                return true;
+            } else if (menuItem.getItemId() == R.id.li_action_mark_read) {
+                markRead(item);
+                return true;
+            } else if (menuItem.getItemId() == R.id.li_action_delete) {
+
+                DialogInterface.OnClickListener dialogClickListener = new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(final DialogInterface dialog, int which) {
+                        switch (which) {
+                            case DialogInterface.BUTTON_POSITIVE:
+                                deleteMessage(item, isParentItem);
+                                dialog.dismiss();
+                                break;
+
+                            case DialogInterface.BUTTON_NEGATIVE:
+                                dialog.dismiss();
+                                break;
+                            default:
+                                break;
                         }
+                    }
+                };
+                AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+                builder.setMessage(R.string.li_are_you_sure)
+                        .setPositiveButton(R.string.li_yes, dialogClickListener)
+                        .setNegativeButton(R.string.li_no, dialogClickListener).show();
+
+                return true;
+            } else if (menuItem.getItemId() == R.id.li_action_subscribe) {
+
+                boolean isSubscribed = item.getUserContext().isSubscribed();
+                if (isSubscribed) {
+                    unsubscribe(item, position);
+                } else {
+                    subscribe(item, position);
+                }
```

Comment:
unnecessary blank line. Through out code please make sure formatting is done correctly. This is Lithium's public facing code

Reply:
sure

Reply:
afair.. Aditya had created a styles guide for the project.. you can use it to ensure the styling matches the res of the project

Reply:
It is updated already, and check-style is run through before committing. Didn't see this issue, anyways updated